scripts/travis/build.sh

```diff
@@ -49,6 +49,17 @@ else
   EXTRA_ARGS=''
 fi
 
+# For PR builds, Skip module-specific tests if its module dependencies haven't been touched
+if [ "$TRAVIS_PULL_REQUEST" != 'false' ]; then
+  # For r2-int-test...
+  # TODO: For now, the list of dependencies is hard-coded into this regex, but ideally it would be dynamic.
+  # TODO: We should run all testing if any *.gradle file is modified, just to be safe.
+  if ! (git diff ${TRAVIS_BRANCH}...HEAD --name-only | grep '^\(data\|entity-stream\|li-protobuf\|log-test-config\|pegasus-common\|r2-core\|r2-disruptor\|r2-filter-compression\|r2-jetty\|r2-netty\|r2-sample\|test-util\)/'); then
+    echo "None of r2-int-test's module dependencies have been touched, skipping tests for r2-int-test..."
```

Comment:
Please add a comment in r2-int-test build.gradle to update this file if a new module dependency is added.

Reply:
Good point, this could get messy if this gets out-of-sync with the actual dependencies. Ideally, we can build this regex dynamically in the build to ensure that it's up-to-date, however I didn't wanna spend time on that right now.

The only issue with this is that it would add time to the build which I think is marginal if it's only for `r2-int-test`. Though if we start doing it for many modules it can become a problem...

Reply:
I think this is okay for now, make sure owners of r2 are aware of this.